packages/devtools_app/lib/src/debugger/evaluate.dart

```diff
@@ -349,3 +283,69 @@ class _ExpressionEvalFieldState extends State<ExpressionEvalField>
     });
   }
 }
+
+Future<List<String>> autoCompleteResultsFor(
+  EditingParts parts,
+  DebuggerController controller,
+) async {
+  final result = <String>{};
+  if (!parts.isField) {
+    result.addAll(controller.variables.value
+        .map((variable) => variable.boundVar.name)
+        .where((name) => name.startsWith(parts.activeWord)));
+  } else {
+    try {
+      var left = parts.leftSide.split(' ').last;
+      // Removing trailing `.`.
+      left = left.substring(0, left.length - 1);
+      final response = await controller.evalAtCurrentFrame(left);
+      if (response is InstanceRef) {
+        final Instance instance = await controller.getObject(response);
+        result.addAll(
+          await _autoCompleteMembersFor(
+            instance.classRef,
+            controller,
+          ),
+        );
+        // TODO(grouma) - This shouldn't be necessary but package:dwds does
+        // not properly provide superclass information.
+        result.addAll(instance.fields.map((field) => field.decl.name));
+        result.removeWhere((prop) => !prop.startsWith(parts.activeWord));
+      }
+    } catch (_) {}
+  }
+  return result.toList();
+}
+
+Future<List<String>> _autoCompleteMembersFor(
+    ClassRef classRef, DebuggerController controller) async {
+  final result = <String>[];
+  if (classRef != null) {
+    final Class clazz = await controller.getObject(classRef);
+    result.addAll(clazz.fields.map((field) => field.name));
+    result.addAll(clazz.functions
+        .where((funcRef) => _validFunction(funcRef, clazz))
+        // The VM shows setters as `<member>=`.
+        .map((funcRef) => funcRef.name.replaceAll('=', '')));
+    result.addAll(await _autoCompleteMembersFor(clazz.superClass, controller));
+    result.removeWhere((member) => !_isAccessible(member, clazz, controller));
+  }
+  return result;
+}
+
+bool _validFunction(FuncRef funcRef, Class clazz) {
+  return !funcRef.isStatic &&
+      !_isContructor(funcRef, clazz) &&
+      funcRef.name != '==';
```

Comment:
need to detect other operators or you will suggest them as invalid autocompletes because the syntax is
`foo+bar` not `foo.+bar`